astropy/coordinates/transformations.py

```diff
@@ -70,6 +70,20 @@ def frame_set(self):
 
         return self._cached_frame_set.copy()
 
+    @property
+    def frame_attrnames_set(self):
+        """
+        A `frozenset` of all the attribute names of all frame classes in this `TransformGraph`.
+        """
+        if self._cached_frame_attrnames_set is None:
+            result = set()
+            for frame_cls in self.frame_set:
+                for attr in frame_cls.get_frame_attr_names().keys():
```

Comment:
Class method get_frame_attr_names() doesn't get the frame attr names, instead it builds a dictionary of the names and attributes: going through the list of attribute names and getattr'ing each one.

The documentation says that it returns the names of attributes and their default values (which is not what you would expect from the method name).

HOWEVER: almost everywhere it is used in astropy, it is used as a list/set of attribute names.  There may be worth in defining get_frame_attr_nameset() that just returns a set of names, instead of an OrderedDict of names + values.